### ISSUE TITLE
[regression/8.0.3] TapGestureRecognizer not working with TableView

### ISSUE THREAD

**Issue by pmoran2550** [dotnet/maui#19348]
### Description

After upgrading a .Net Maui project from .Net 7 to .Net 8 my tap gestures within a tableview were no longer working. I created a sample project to reproduce the issue. I am seeing this using the Android simulator (Android 13.0 - API 33) and IOS simulator (iPhone 15 iOS 17).

### Steps to Reproduce

Clone project https://github.com/pmoran2550/GestureTapIssue.git
Build and run
Taping on the Label and StackPanel show the alert called when the tap is recognized. Taping on the table items do not.

### Link to public reproduction project repository

https://github.com/pmoran2550/GestureTapIssue.git

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

7.0.101

### Affected platforms

iOS, Android

### Affected platform versions

Android 13, iOS 17

### Did you find any workaround?

No

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.9.0 Preview 2(8.0.3). Repro on Windows 11, Android 14.0-API34, iOS 17.0 and MacCatalyst with below Project:
[MauiAppTest.zip](https://github.com/dotnet/maui/files/13916984/MauiAppTest.zip)
![Windows](https://github.com/dotnet/maui/assets/97647558/9534a1bd-f359-46b9-aecc-9177335dc9b4)

**Comment by FlavioGoncalves-Cayas** [comment]
Something I don't understand is that it works with 8.0.6-nightly.9851 which was built before SR1 was released. But somehow with SR1 it does no longer work. So is it a regression or is the fix for some reason not included in SR1?

**Comment by Redth** [comment]
The versioning is a bit confusing since we had nightlies building as 8.0.6-nightly.xyz after we branched for SR1 which ended up shipping with version 8.0.6 (originally it was going to be 8.0.5 but we found a regression last minute that we fixed causing the version to bump).  We'll take a look and see if we can make some changes to nightly versioning to make that less confusing.

SR1 built Dec 7, and the PR with the fix for this (https://github.com/dotnet/maui/pull/18938) merged on Dec 11, _after_ the branching happened, so this fix is going into SR2.
